applications/asset_tracker_v2/src/cloud/lwm2m_integration/lwm2m_integration.c

```diff
@@ -206,9 +206,58 @@ static int device_reboot_cb(uint16_t obj_inst_id, uint8_t *args, uint16_t args_l
 	return 0;
 }
 
+/* Callback handler triggered when the modem should be put in a certain functional mode.
+ * Handler is called pre provisioning of DTLS credentials when the modem should be put in
+ * offline mode, and when the modem should return to normal mode after
+ * provisioning has been carried out.
+ */
+static int modem_mode_request_cb(enum lte_lc_func_mode new_mode, void *user_data)
+{
+	ARG_UNUSED(user_data);
+
+	int err;
+	enum lte_lc_func_mode mode_current;
+	struct cloud_wrap_event cloud_wrap_evt = { 0 };
+
+	err = lte_lc_func_mode_get(&mode_current);
+	if (err) {
+		LOG_ERR("lte_lc_func_mode_get failed, error: %d", err);
```

Comment:
The errors from `lte_lc_func_mode_get` seem to only occur when calling with an invalid mode. As far as I understand, the mode comes from the lwm2m library and not the application developer. Would an `__ASSERT` increase the likelihood of us catching the error?

Reply:
`lte_lc_func_mode_get` is  a function that populates `mode_current` with the current mode of the modem. So whatever `mode_current` is prior to the call it shouldn't cause any run-time issue. And if the passed in pointer type is changed a build time error will be triggered on a type mismatch.

If `lte_lc_func_mode_get`  fails it means that there is some error in the underlying calls to `nrf_modem_at_scanf` when retrieving the mode. This can for instance happen if the modem is in some erroneous state.

Ultimately, if `lte_lc_func_mode_get` fails a global graceful shutdown is triggered in the application. If we were to use an assert we would loose this and reboot at once.

Reply:
Ok, then I agree an error is the best way to handle it.